### PR TITLE
Add licenses

### PR DESCRIPTION
**Pull request by markuskowa**
###### Motivation for this change
Another round for https://github.com/NixOS/nixpkgs/issues/43716

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libgnome-keyring/default.nix [hunk 1]
```diff
@@ -16,5 +16,6 @@ stdenv.mkDerivation {
 
   meta = {
     inherit (glib.meta) platforms maintainers;
+    license = stdenv.lib.licenses.gpl2;
```
[the remark this answers is not in the document]
Comment (markuskowa) [reply pkgs/development/libraries/libgnome-keyring/default.nix]: I usually go by the COPYING/LICENSE files.

File: pkgs/development/libraries/libdbusmenu-qt/qt-5.5.nix [hunk 2]
```diff
@@ -19,5 +19,6 @@ stdenv.mkDerivation {
     description = "Provides a Qt implementation of the DBusMenu spec";
     maintainers = [ maintainers.ttuegel ];
     inherit (qtbase.meta) platforms;
+    license = licenses.gpl2;
```
[the remark this answers is not in the document]
Comment (markuskowa) [reply pkgs/development/libraries/libdbusmenu-qt/qt-5.5.nix]: I'll fix it in the next round.

File: pkgs/development/libraries/libdbusmenu-qt/default.nix [hunk 3]
```diff
@@ -23,7 +23,7 @@ stdenv.mkDerivation {
   meta = with stdenv.lib; {
     description = "Provides a Qt implementation of the DBusMenu spec";
     inherit homepage;
-    maintainers = [ ];
     inherit (qt4.meta) platforms;
+    license = licenses.gpl2;
```
[the remark this answers is not in the document]
Comment (c0bw3b) [reply pkgs/development/libraries/libdbusmenu-qt/default.nix]: You're right it's LGPL2 only for this one

File: pkgs/development/libraries/libiodbc/default.nix [hunk 4]
```diff
@@ -17,9 +17,10 @@ stdenv.mkDerivation rec {
       export NIX_LDFLAGS_BEFORE="-rpath $out/lib"
     '';
 
-  meta = {
+  meta = with stdenv.lib; {
     description = "iODBC driver manager";
     homepage = http://www.iodbc.org;
-    platforms = stdenv.lib.platforms.unix;
+    platforms = platforms.unix;
+    license = licenses.bsd3;
```
[the remark this answers is not in the document]
Comment (c0bw3b) [reply pkgs/development/libraries/libiodbc/default.nix]: No we don't _need_ it per se but the fact that we have a list for `meta.license` allows us to expose both to Nix/NixOS users and let them choose which one suits their requirements.
And I think this is what might be intended upstream with a dual licensing.

[the remark this answers is not in the document]
Comment (markuskowa) [reply pkgs/development/libraries/libiodbc/default.nix]: The meaning of a list of licenses is not clearly defined. One could interpret it as dual licensing ("pick one") or the collection different licenses used within the package ("all apply", what it means in the majority of cases). To make it even more complicated: a lot of libraries come with a set tools and the library has for example a LGPL license and the tools have  a GPL license. 
There is no way of telling what the list means just by looking at `meta.license` alone.

The reason why I picked one was an earlier discussion with a similar dual licensing constellation: https://github.com/NixOS/nixpkgs/pull/45282#discussion_r211084863

[the remark this answers is not in the document]
Comment (c0bw3b) [reply pkgs/development/libraries/libiodbc/default.nix]: Ah thanks for the link to the previous discussion.
The interpretation of the meta.licenses list remains unclear indeed. Is it OR, AND
Myself I read it as "_here is all the licenses that applies to parts or totality of this project_".
Then it's up to the user to dig deeper if he needs/wants to.

But just like joachifm in the previous PR, I was just raising a question but I'm fine either way. There is no definitive good/bad policy there. :)

[the remark this answers is not in the document]
Comment (markuskowa) [reply pkgs/development/libraries/libiodbc/default.nix]: I don't have strong opinion on this topic but I would tend to stay with picking the more flexible license for the sake of simplicity.

**Pull request merged**